src/main/java/net/shadowmage/ancientwarfare/npc/entity/faction/attributes/EntityVehicleProperty.java

```diff
@@ -0,0 +1,48 @@
+package net.shadowmage.ancientwarfare.npc.entity.faction.attributes;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.storage.loot.properties.EntityProperty;
+import net.shadowmage.ancientwarfare.npc.AncientWarfareNPC;
+import net.shadowmage.ancientwarfare.npc.entity.vehicle.IVehicleUser;
+import net.shadowmage.ancientwarfare.vehicle.entity.VehicleBase;
+
+import java.util.Random;
+
+public class EntityVehicleProperty implements EntityProperty {
+	private final String testVehicle;
+
+	private EntityVehicleProperty(String testVehicle) {
+		this.testVehicle = testVehicle;
+	}
+
+	public boolean testProperty(Random random, Entity entityIn) {
+		if (entityIn instanceof IVehicleUser && (((IVehicleUser) entityIn).isRidingVehicle())) {
```

Comment:
After checking the entity instance type, cast the entity into new IVehicleUser variable and reuse that variable in all subsequent uses where there's currently ((IVehicleUser) entityIn). Makes the code more readable (if there was just one use of this cast that would sound good to me to directly use it but there are three so it should help with readability in this case).

Reply:
Made the changes, take a look please.